client/my-sites/checkout/composite-checkout/composite-checkout.js

```diff
@@ -624,6 +628,30 @@ function useCountryList( overrideCountryList ) {
 	return countriesList;
 }
 
+function useRecordCheckoutLoaded(
+	recordEvent,
+	isApplePayAvailable,
+	responseCart,
+	storedCards,
+	isLoadingStoredCards,
+	product
+) {
+	const hasRecordedCheckoutLoad = useRef( false );
+	if ( ! isLoadingStoredCards && ! hasRecordedCheckoutLoad.current ) {
```

Comment:
I never got around to adding error handling for loading stored cards. I suggest maybe adding something like the following patch so that this can be triggered even if loading stored cards fails:

```diff
--- a/client/my-sites/checkout/composite-checkout/payment-method-helpers.js
+++ b/client/my-sites/checkout/composite-checkout/payment-method-helpers.js
@@ -17,7 +17,7 @@ import {

 const debug = debugFactory( 'calypso:composite-checkout:payment-method-helpers' );

-export function useStoredCards( getStoredCards ) {
+export function useStoredCards( getStoredCards, onEvent ) {
        const [ state, dispatch ] = useReducer( storedCardsReducer, {
                storedCards: [],
                isLoading: true,
@@ -29,14 +29,19 @@ export function useStoredCards( getStoredCards ) {
                        return getStoredCards();
                }

-               // TODO: handle errors
-               fetchStoredCards().then( ( cards ) => {
-                       debug( 'stored cards fetched', cards );
-                       isSubscribed && dispatch( { type: 'FETCH_END', payload: cards } );
-               } );
+               fetchStoredCards()
+                       .then( ( cards ) => {
+                               debug( 'stored cards fetched', cards );
+                               isSubscribed && dispatch( { type: 'FETCH_END', payload: cards } );
+                       } )
+                       .catch( ( error ) => {
+                               debug( 'stored cards failed to load', error );
+                               onEvent( { type: 'STORED_CARD_ERROR', payload: error.message } );
+                               isSubscribed && dispatch( { type: 'FETCH_END', payload: [] } );
+                       } );

                return () => ( isSubscribed = false );
-       }, [ getStoredCards ] );
+       }, [ getStoredCards, onEvent ] );
        return state;
 }

diff --git a/client/my-sites/checkout/composite-checkout/record-analytics.js b/client/my-sites/checkout/composite-checkout/record-analytics.js
index a90c6f22ae..7fd5236eb3 100644
--- a/client/my-sites/checkout/composite-checkout/record-analytics.js
+++ b/client/my-sites/checkout/composite-checkout/record-analytics.js
@@ -113,6 +113,14 @@ export default function createAnalyticsEventHandler( reduxDispatch ) {
                                        } )
                                );

+                       case 'STORED_CARD_ERROR':
+                               return reduxDispatch(
+                                       recordTracksEvent( 'calypso_checkout_composite_stored_card_error', {
+                                               error_type: action.payload.type,
+                                               error_message: String( action.payload.message ),
+                                       } )
+                               );
+
                        case 'CART_ERROR':
                                return reduxDispatch(
                                        recordTracksEvent( 'calypso_checkout_composite_cart_error', {
```